native/src/components/DatePicker.tsx

```diff
@@ -0,0 +1,155 @@
+import { DateTime } from 'luxon'
+import React, { ReactElement, useEffect, useState } from 'react'
+import styled from 'styled-components/native'
+
+import { CalendarTodayIcon } from '../assets'
+import Icon from './base/Icon'
+import IconButton from './base/IconButton'
+import Text from './base/Text'
+
+const DateContainer = styled.View`
+  width: fit-content;
+  position: relative;
+`
+const Input = styled.TextInput`
+  text-align: center;
+`
+const StyledInputWrapper = styled.View`
+  min-width: 316px;
+  height: 56px;
+  padding: 0 16px;
+  border-radius: 8px;
+  border-color: ${props => props.theme.colors.themeColorLight};
+  border-width: 3px;
+  border-style: solid;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+`
+const Wrapper = styled.View`
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+`
+const StyledIconButton = styled(IconButton)<{ $isModalOpen: boolean }>`
+  width: 40px;
+  height: 40px;
+  background-color: ${props =>
+    props.$isModalOpen ? props.theme.colors.themeColorLight : props.theme.colors.textDisabledColor};
+`
+const StyledTitle = styled.Text`
+  background-color: ${props => props.theme.colors.backgroundColor};
+  position: absolute;
+  top: -12px;
+  left: 12px;
+  right: auto;
+  padding: 2px 5px;
+  font-size: 12px;
+  font-weight: 400;
+  z-index: 1;
+`
+const StyledError = styled.Text`
+  font-size: 12px;
+  font-weight: bold;
+  color: ${props => props.theme.colors.invalidInput};
+`
+export type DatePickerProps = {
+  title: string
+  date: DateTime | null
+  setDate: (date: DateTime | null) => void
+  error?: string
+  modalOpen: boolean
+  setModalOpen: (open: boolean) => void
+}
+const DatePicker = ({ title, date, setDate, error, modalOpen, setModalOpen }: DatePickerProps): ReactElement => {
+  const [inputDay, setInputDay] = useState(date?.toFormat('dd'))
+  const [inputMonth, setInputMonth] = useState(date?.toFormat('MM'))
+  const [inputYear, setInputYear] = useState(date?.toFormat('yyyy'))
```

Comment:
I think it is not very good, to have separate state for day, month and year. This leads to
1) unnecessary re-renders
2) allows invalid dates to be entered, like 31.02.2024

Reply:
> I think it is not very good, to have separate state for day, month and year. This leads to
> 
>     1. unnecessary re-renders

Tbh I don't think there is a difference between separate states or single state for the date here. If one value is changed, there is a rerender. It does not matter if we change a single state multiple times or a multi state multiple times.

>     2. allows invalid dates to be entered, like 31.02.2024

It is never completely preventable that no invalid date is entered, for example by copy and paste. If it is correctly handled/validated, I don't see a reason why that has to be a problem.